### DISCUSSION TITLE
market.getChart() possibility

### DISCUSSION THREAD

**Discussion by edzebo** [Mathieu2301/TradingView-API#40] [Q&A] [answered]
Hi,

First of all, thank you very much for your work and for making this open-source utility.

Now, onto the question. 
I tried figuring out and adjusting code to get existing chart data but was not able to get it working so far or I am missing something obvious.
For example, I have a chart set up that is public with multiple indicators. Is it possible somehow to give chartId or chartLayout string and based on that pull in the data including indicator values and also price values?

Thank you very much!

**Comment by Mathieu2301** [comment, marked as answer]
Hello, it's currently not possible if you don't know the market symbol and indicators IDs.
But yes, we can imagine a `getLayout(layoutID, sessionid = '')` function to get the maket, the indicators, etc... (`sessionid` is only required for private layouts).

All this data can be fetched by getting source of `https://www.tradingview.com/chart/${layoutID}/` and parsing the JSON from the following Regex expression result: `/initData\.content = (.*);\n/i`.

**Comment by edzebo** [comment]
Got it. Thank you.
